theano/gof/op.py

```diff
@@ -694,6 +694,9 @@ def __ne__(self, other):
     # Python implementation #
     #########################
 
+    def grad2(self, inputs, outputs, output_grads):
```

Comment:
Would it be possible to either:
- use a metaclass in Op so that would automatically redefine the `grad` method so it accepts (and ignore) `outputs`, or
- use `inspect` in `theano.gradient` to see what it accepts and pass that (rather than try/except)?